src/main/java/com/microfocus/application/automation/tools/octane/events/OutputEnvironmentParametersHelper.java

```diff
@@ -0,0 +1,80 @@
+package com.microfocus.application.automation.tools.octane.events;
+
+import com.microfocus.application.automation.tools.settings.OctaneBuildWrapper;
+import com.microfocus.application.automation.tools.settings.RunnerMiscSettingsGlobalConfiguration;
+import com.microfocus.application.automation.tools.sse.common.StringUtils;
+import hudson.EnvVars;
+import hudson.model.AbstractBuild;
+import hudson.model.BuildableItemWithBuildWrappers;
+import hudson.model.Job;
+import hudson.model.Run;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class OutputEnvironmentParametersHelper {
+
+	public static final String SPLIT_SYMBOL = " ";
+
+	public static Map<String, String> getOutputEnvironmentParams(Run run) {
+		EnvVars environment = getEnvironment(run);
+		if (environment == null) {
+			return Collections.emptyMap();
+		} else {
+			List<String> paramKeysList = new ArrayList<>();
+			paramKeysList.addAll(getGlobalParamsList());
+			paramKeysList.addAll(getJobParamsList(run));
+
+			if (paramKeysList.isEmpty()) return Collections.emptyMap();
+
+			Map<String, String> outputEnvParams = new HashMap<>();
+			Set<String> sensitiveBuildVariables =null;
+			if (run instanceof AbstractBuild) {
+				sensitiveBuildVariables = ((AbstractBuild) run).getSensitiveBuildVariables();
+			}
+
+			String value;
+			for (String key : paramKeysList) {
+				if (sensitiveBuildVariables != null && sensitiveBuildVariables.contains(key)) continue;
+				value = environment.get(key);
+				if (value != null) {
+					outputEnvParams.put(key, value);
+				}
+			}
+			return outputEnvParams;
+		}
+	}
+
+	private static EnvVars getEnvironment(Run run) {
+		EnvVars environment = null;
+		try {
+			environment = run.getEnvironment(null);
+		} catch (IOException | InterruptedException ignored) {
```

Comment:
Can you please add to plugin about error here, no need all exception log , can put general message and add exception message

Reply:
added, please verify the message